### ISSUE TITLE
DeleteBinObj bug with AndroidResource

### ISSUE THREAD

**Issue by jonathanpeppers** [xamarin/xamarin-android#2193]
### Steps to Reproduce

1. `File | New Project | Android App (Xamarin) | Tabbed App`, set `Minimum Android Version` to `Android 5.0 (Lollipop)`
2. Build
3. Open `Resource\layout\activity_main.axml` (I used the ancient VS XML editor instead of the designer, to remove that from the equation)
4. Add a new `TextView`
5. Run via F5 or the Play button

A crash at runtime is encountered:
```
Unhandled Exception:

Android.Views.InflateException: Binary XML file line #1: Binary XML file line #1: Error inflating class android.support.design.widget.BottomNavigationView occurred
```

On this line:
```csharp
SetContentView (Resource.Layout.activity_main);
```
My `TextView` source is basically the default if dragged from the toolbox in the designer:
```xml
  <TextView
      android:text="Text"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      android:minWidth="25px"
      android:minHeight="25px"
      android:id="@+id/textView1" />
```

Once in this state, a `Rebuild` fixes the problem. You can deploy again to see the issue resolved.

### Expected Behavior

Modifying a layout should not cause a runtime exception!

### Actual Behavior

Modifying a layout causes runtime exception! `Rebuild` resolves the issue...

### Version Information

I have been able to repro this on VS 2017 15.8.4, and the build of Xamarin.Android we will be shipping with 15.9 P3.

### Log File

Relevant logs: [DeleteBinObj.zip](https://github.com/xamarin/xamarin-android/files/2394164/DeleteBinObj.zip)

> VS bug [#686581](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/686581)

**Comment by jonathanpeppers** [comment]
Hmm, weird `classes.dex` has the type from the error message!

```
obj\Debug\MonoAndroid81\android\bin> dexdump.exe .\classes.dex | Select-String 'Landroid/support/design/widget/BottomNavigationView;'

type          : 'Landroid/support/design/widget/BottomNavigationView;'
```

So perhaps `obj\Debug\MonoAndroid81\android\bin\packaged_resources` is the file that's messed up?

I'll keep digging...

**Comment by jonathanpeppers** [comment]
@dellis1972 repro project here: https://github.com/jonathanpeppers/DeleteBinObj

My commands were:
```
git clean -dxf
msbuild .\DeleteBinObj\DeleteBinObj.csproj /bl:build.binlog /restore
git stash pop
msbuild .\DeleteBinObj\DeleteBinObj.csproj /bl:install.binlog /t:Install
```

Instead of `git stash pop` you can just add this to `Resource\layout\activity_main.axml`:
```xml
  <TextView
      android:text="Text"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      android:minWidth="25px"
      android:minHeight="25px"
      android:id="@+id/textView1" />
```

**Comment by Silic0nS0ldier** [comment]
If anyone hits this a work around is to add
```xml
<Target Name="BeforeBuild" Condition="Exists('$(ProjectDir)obj\Debug\MonoAndroid81\android\src\android\support\compat\R.java')">
  <Delete Files="$(ProjectDir)obj\Debug\MonoAndroid81\android\src\android\support\compat\R.java" />
</Target>
```
Or something similar (`MonoAndroid81` might be different) to the project file.

**Comment by Silic0nS0ldier** [comment]
Fix is out in Visual Studio 15.9 Preview 4, and I can confirm it works.
